src/components/org/apache/jmeter/extractor/json/render/RenderAsJsonRenderer.java

```diff
@@ -0,0 +1,259 @@
+import javax.swing.JTabbedPane;
+import javax.swing.JTextArea;
+import javax.swing.border.Border;
+import javax.swing.border.EmptyBorder;
+
+import org.apache.jmeter.extractor.json.jsonpath.JSONManager;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jmeter.visualizers.RenderAsJSON;
+import org.apache.jmeter.visualizers.ResultRenderer;
+import org.apache.jmeter.visualizers.ViewResultsFullVisualizer;
+import org.apache.jorphan.gui.GuiUtils;
+import org.apache.jorphan.gui.JLabeledTextField;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.log.Logger;
+
+
+/**
+ * Implement ResultsRender for JSON Path tester
+ * @since 2.14
+ */
+public class RenderAsJsonRenderer implements ResultRenderer, ActionListener {
+
+    private static final Logger LOGGER = LoggingManager.getLoggerForClass();
+
+    private static final String TAB_SEPARATOR = "    "; //$NON-NLS-1$
+    
+    private static final String JSONPATH_TESTER_COMMAND = "jsonpath_tester"; // $NON-NLS-1$
+
+    private JPanel jsonWithJSonPathPane;
+
+    private JTextArea jsonDataField;
+
+    private JLabeledTextField jsonPathExpressionField;
+
+    private JTextArea jsonPathResultField;
+
+    private JTabbedPane rightSide;
+
+    private SampleResult sampleResult = null;
+
+    private JScrollPane jsonDataPane;
+
+
+    /** {@inheritDoc} */
+    public void clearData() {
+        this.jsonDataField.setText(""); // $NON-NLS-1$
+        // don't set empty to keep json path
+        this.jsonPathResultField.setText(""); // $NON-NLS-1$
+    }
+
+    /** {@inheritDoc} */
+    public void init() {
+        // Create the panels for the json tab
+        jsonWithJSonPathPane = createJSonPathExtractorPanel();
+    }
+
+    /**
+     * Display the response as text or as rendered HTML. Change the text on the
+     * button appropriate to the current display.
+     *
+     * @param e the ActionEvent being processed
+     */
+    public void actionPerformed(ActionEvent e) {
+        String command = e.getActionCommand();
+        if ((sampleResult != null) && (JSONPATH_TESTER_COMMAND.equals(command))) {
+            String response = jsonDataField.getText();
+            executeAndShowXPathTester(response);
+        }
+    }
+
+    /**
+     * Launch json path engine to parse a input text
+     * @param textToParse
+     */
+    private void executeAndShowXPathTester(String textToParse) {
+        if (textToParse != null && textToParse.length() > 0
+                && this.jsonPathExpressionField.getText().length() > 0) {
+            this.jsonPathResultField.setText(process(textToParse));
+            this.jsonPathResultField.setCaretPosition(0); // go to first line
+        }
+    }
+
+    private String process(String textToParse) {
+        try {
+            List<String> matchStrings = new ArrayList<String>();
+            extractWithJSonPath(textToParse, jsonPathExpressionField.getText(), matchStrings);
```

Comment:
Why do we pass in a list instead of creating it inside of `extractWithJsonPath` and returning it?